### ISSUE TITLE
Issues with `vue/no-unused-components` rule: wrong report

### ISSUE THREAD

**Issue by tinymins** [vuejs/eslint-plugin-vue#768]
**Tell us about your environment**	
* **ESLint version:**	5.12.0
* **eslint-plugin-vue version:**	5.1.0
* **Node version:**	8

**Please show your full configuration:**

```json
// Default
```

**What did you do?**	

```vue
import XButton from 'path/to/x-button';

export default {
  components: {
    [XButton.name]: XButton,
  },
};
```

**What did you expect to happen?**
Should not report error.

**What actually happened?**

  ✘  https://google.com/#q=vue%2Fno-unused-components  The "name" component has been registered but not used
  src/views/secret/create.vue:219:5

**Comment by ota-meshi** [comment]
@tinymins Thank you for this issue.
I will fix this issue.
